app/daos/UserDaoMongoDBImpl.js

```diff
@@ -22,7 +22,20 @@ module.exports = function() {
     });
   };
 
+  const signin = function(userData) {
```

Comment:
I _think_ I know what you're trying to do. This method should be simple since it's a DAO method, and should probably just be `getByEmail` with `email` being the argument. You could also create `getByUsername`, and also have a combined method `getByEmailOrUsername` that tries to support both.